pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -535,6 +542,42 @@ protected void handleAuthResponse(CommandAuthResponse authResponse) {
         try {
             AuthData clientData = AuthData.of(authResponse.getResponse().getAuthData());
             doAuthentication(clientData);
+            if (connectionPool != null && state == State.ProxyLookupRequests) {
+                if (service.getConfiguration().isForwardAuthorizationCredentials()) {
+                    connectionPool.getPool().values().forEach(n -> n.values().forEach(cnxFuture -> {
+                        String clientVersion;
+                        if (authResponse.hasClientVersion()) {
+                            clientVersion = authResponse.getClientVersion();
+                        } else {
+                            clientVersion = PulsarVersion.getVersion();
+                        }
+                        int protocolVersion;
+                        if (authResponse.hasProtocolVersion()) {
+                            protocolVersion = authResponse.getProtocolVersion();
+                        } else {
+                            protocolVersion = Commands.getCurrentProtocolVersion();
+                        }
+
+                        ByteBuf cmd =
+                                Commands.newAuthResponse(clientAuthMethod, clientData, protocolVersion, clientVersion);
+                        cnxFuture.thenAccept(clientCnx -> clientCnx.ctx().writeAndFlush(cmd)
```

Comment:
I think we need some logs if the `cnxFuture` is completed with exception

Reply:
And the name is also confusing, it should be `toBrokerCnx`?

Reply:
> And the name is also confusing, it should be `toBrokerCnx`?

Yes.